daemon/create.go

```diff
@@ -47,7 +48,7 @@ func (daemon *Daemon) ContainerCreate(job *engine.Job) engine.Status {
 
 	container, buildWarnings, err := daemon.Create(config, hostConfig, name)
 	if err != nil {
-		if daemon.Graph().IsNotExist(err) {
+		if daemon.Graph().IsNotExist(err) && strings.Contains(err.Error(), config.Image) {
```

Comment:
Maybe we can fix `IsNotExist` signature. Becuase it was such pain before, you found I think sixth bug about it.
What about `IsNotExist(err, id)`? WDYT?